lib/src/web_socket/common/w_socket.dart

```diff
@@ -126,31 +103,58 @@ abstract class CommonWSocket extends Stream implements WSocket {
   /// Sending additional data before this stream has completed may
   /// result in a [StateError].
   Future addStream(Stream stream) async {
-    await sink.addStream(stream);
+    return _outgoing.addStream(stream);
   }
 
   /// Closes the WebSocket connection. Optionally set [code] and [reason]
   /// to send close information to the remote peer.
   Future close([int code, String reason]) {
-    if (_isClosed) return done;
-    _isClosed = true;
     shutDown(code: code, reason: reason);
     return done;
   }
 
-  void closeSocket(int code, String reason);
+  StreamSubscription listen(void onData(event),
+      {Function onError, void onDone(), bool cancelOnError}) {
+    var sub = _incoming.stream
+        .listen(onData, onError: onError, cancelOnError: cancelOnError);
+    _incomingSubscription = new WSocketSubscription(sub, onDone, onCancel: () {
+      _incomingSubscription = null;
```

Comment:
The code path isn't very clear here, but this `onCancel` callback only gets called when `WSocketSubscription.cancel()` is called, so that would have already happened (https://github.com/Workiva/w_transport/pull/142/files#diff-8abde7635f40c28382f006d30214b7cdR14)